File: src/main/java/com/sofodev/thedragonlib/util/ArrayUtils.java
package com.sofodev.thedragonlib.util;

import java.util.Arrays;
import java.util.Locale;

public class ArrayUtils {

    public static String[] arrayToLowercase(String[] array) {
        return Arrays.stream(array).map(s -> s.toLowerCase(Locale.ENGLISH)).toArray(String[]::new);
    }

    /**
     * Shift all objects in an array by the number of places specified.
     * <p/>
     * Example:
     * Object[] array = {1, 2, 3, 4, 5}
     * array = arrayShift(array, 1);
     * array now equals {5, 1, 2, 3, 4}
     * Shift can be a positive or negative number
     * <p/>
     * Shift is not restricted to any max or min value so it could for example be linked to a counter that
     * continuously increments.
     */
    public static Object[] arrayShift(Object[] input, int shift) {
        Object[] newArray = new Object[input.length];

        for (int i = 0; i < input.length; i++) {
            int newPos = (i + shift) % input.length;

            if (newPos < 0) newPos += input.length;

            newArray[newPos] = input[i];
        }

        return newArray;
    }
}

File: src/main/java/net/thedragonteam/thedragonlib/util/ArrayUtils.kt
package net.thedragonteam.thedragonlib.util

import java.util.*

object ArrayUtils {

    fun arrayToLowercase(array: Array<out String>): Array<String?> {
        val lowercaseArray = arrayOfNulls<String>(array.size)
        for (i in array.indices) {
            lowercaseArray[i] = array[i].toLowerCase(Locale.ENGLISH)
        }
        return lowercaseArray
    }

    /**
     * Shift all objects in an array by the number of places specified.
     *
     *
     * Example:
     * Object[] array = {1, 2, 3, 4, 5}
     * array = arrayShift(array, 1);
     * array now equals {5, 1, 2, 3, 4}
     * Shift can be a positive or negative number
     *
     *
     * Shift is not restricted to any max or min value so it could for example be linked to a counter that
     * continuously increments.
     */
    fun arrayShift(input: Array<Any>, shift: Int): Array<Any?> {
        val newArray = arrayOfNulls<Any>(input.size)

        for (i in input.indices) {
            var newPos = (i + shift) % input.size

            if (newPos < 0) {
                newPos += input.size
            }

            newArray[newPos] = input[i]
        }

        return newArray
    }
}